compiler/src/iree/compiler/DispatchCreation/FuseHorizontalContractions.cpp

```diff
@@ -47,62 +48,69 @@ struct FuseHorizontalContractionsPass final
 
 } // namespace
 
-/// Structs that captures the ops that are to be fused
-struct HorizontalFusionGroup {
-  // Contractions op that are to be fused.
-  SmallVector<linalg::LinalgOp> contractionOps;
-  // Optional truncate operations that could be following the contraction op.
-  std::optional<SmallVector<linalg::GenericOp>> truncateOps;
-};
-
 /// Helper method to check operations equivalence
-static bool checkOperationEquivalence(Operation *lhsOp, Operation *rhsOp) {
-  // During equivalence check, it would have been easier if `checkEquivalence`
-  // would just use `OpOperands *`. Since it takes `Value`s for now, just
-  // check that the values are the same as operands. This is potentially
-  // making the match too broad, but is an OK work-around for now.
-  // TODO(MaheshRavishankar): Fix upstream `checkEquivalence` signater in
-  // `OperationEquivalence::isEquivalentTo`.
-  llvm::SmallDenseSet<Value, 8> operands;
-  operands.insert(lhsOp->operand_begin(), lhsOp->operand_end());
-  operands.insert(rhsOp->operand_begin(), rhsOp->operand_end());
-
-  llvm::DenseMap<Value, Value> equivalentValues;
-  auto checkEquivalent = [&](Value lhsValue, Value rhsValue) {
-    if (operands.contains(lhsValue) && operands.contains(rhsValue)) {
-      return success();
-    }
-    return success(equivalentValues.lookup(lhsValue) == rhsValue ||
-                   equivalentValues.lookup(rhsValue) == lhsValue);
+static bool checkOperationEquivalence(Operation *aOp, Operation *bOp) {
```

Comment:
The function is tied to contraction in the implementation details, and it is assuming the operations all implement linalg::ContractionOpInteface. Perhaps we can rename it as `checkContractionEquivalence`?